ement.el

```diff
@@ -222,6 +216,51 @@ call `pop-to-buffer'."
 
 ;;;; Functions
 
+(defun ement--read-username-and-server (prompt &optional initial-input history
+                                               default-value
+                                               inherit-input-method)
+  "Read Matrix user ID, submitted in the form \"@username:server\".
+
+Return (user-input . plist) where plist is `auth-source-search'-compatible plist."
+  (cl-loop
```

Comment:
I understand why you use the loop, but I think it's not necessary here.  On the rare occasion that a user mistypes the ID, they can call the command again.